src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -3401,11 +3405,19 @@ s32 Player_UpdateUpperBody(Player* this, PlayState* play) {
     return 1;
 }
 
-s32 func_80836898(PlayState* play, Player* this, PlayerFuncA74 func) {
-    this->func_A74 = func;
-    Player_SetupAction(play, this, Player_Action_808458D0, 0);
+/**
+ * This system exists to allow Player to delay a certain function until after an item in hand
+ * gets put away. The function provided by `putAwayDelayFunc` will only run after the put away
+ * process is fully complete.
+ *
+ * @return  From `Player_PutAwayHeldItem`: true if an item needs to be put away, false if not.
+ */
+s32 Player_SetPutAwayDelay(PlayState* play, Player* this, PutAwayDelayFunc putAwayDelayFunc) {
```

Comment:
My first impression is that `SetPutAwayDelay` sounds like it's setting the length of time before an item is put away. Some suggestions:

* action func: `Player_Action_PutAwayAutomatic` (as opposed to `Player_Action_PutAwayManual` for the A button action)
* thing that happens after put away: `AfterPutAwayFunc`/`afterPutAwayFunc`
* setup func: `Player_SetupPutAwayAutomatic` (I notice that these setup functions are not named yet in player, but it fits the naming convention of other actors. I understand that these setup funcs are not 1-to-1 with action funcs so it could get weird, but there are many functions like this whose only job is to set up one action)

Reply:
this is an interesting option to explore. I will first want to make sure that there arent any other "automatic putaways" that can conflict with this name. if there arent then I will go ahead and try this.

Reply:
There are a bunch of other examples of putting away things automatically that dont use this system
- unequipping an item from c buttons
- putting away deku stick if you break it
- putting away an item when opening a door
a few others

I think the main thing here is that this system is handled by a whole action func itself, keeping player from doing literally anything else until the putaway is done. All other examples of automatic putaways can happen simultaneously with other actions. This is what I was trying to convey with the "Delay" naming.

I think this block puts this difference into the best perspective. You can see that if player is on a horse, `Player_PutAwayHeldItem` can run right away, with player simultaneously running the "be on horse" action function.
Meanwhile when swimming or standing on the ground, link uses this delay system to ONLY put away the item before hes allowed to continue with the talking action later
```c
        if (this->stateFlags1 & PLAYER_STATE1_23) {
            s32 sp24 = this->av2.actionVar2;

            Player_PutAwayHeldItem(play, this);
            func_8083A2F8(play, this);

            this->av2.actionVar2 = sp24;
        } else {
            if (func_808332B8(this)) {
                Player_SetPutAwayDelay(play, this, func_8083A2F8);
                Player_AnimChangeLoopSlowMorph(play, this, &gPlayerAnim_link_swimer_swim_wait);
```
I think it would be good if the naming for this system could capture that aspect

Reply:
Hmm I think would it still be accurate to say that this action func is used to put away items automatically before performing another action, even if not all instances of putting away items automatically use this action. But I don't really have any other suggestions for the action name. I do still prefer `afterPutAwayFunc` and `Player_Setup<ActionName>` (or maybe `Player_SetupAction_<ActionName>`) over the current names though.

Reply:
My point here is that it isnt just any old action function, its a pretty unique system to delay some function until after putaway is complete.
The action here, to me, is not putting away the item. That is an Upper Action handled by the upper body system. The action this function is responsible for is delaying the next function (which itself is typically another SetupAction func) until the putaway is done. This is why its named more about the concept of the delay, and not following the usual "setup action" terminology that you would see with a normal action. 

Player will have actions that use the Setup<Action> terminology, but this one to me is not a normal action that needs it. Its more about setting the function that sets up the next action AFTER the delay.

I am still open to the idea of naming it along the lines of what you originally suggested (with some comments explaining some of the things I mention above), however I do think the direction Im trying to go in is best. Summed up, its more of a mechanical operation handled with an action function than it is an actual action like you would find in any actor.

Reply:
If we went with for example `Player_Action_WaitForPutAway`, this would be more aligned with my idea that the action is not the putaway, but the delaying/waiting.
Im not sure `Player_SetupWaitForPutAway` is enough for me to convey the significance of the function pointer/callback involved in the system. But it could be okay (and explained in a comment)

Reply:
> The action here, to me, is not putting away the item. That is an Upper Action handled by the upper body system. The action this function is responsible for is delaying the next function (which itself is typically another SetupAction func) until the putaway is done. This is why its named more about the concept of the delay, and not following the usual "setup action" terminology that you would see with a normal action.

I see, that makes a lot of sense. I like your suggestion of `Player_Action_WaitForPutAway`

> Player will have actions that use the Setup terminology, but this one to me is not a normal action that needs it. Its more about setting the function that sets up the next action AFTER the delay.

Hm I'm not convinced this action func should be treated differently than the rest. The action func state is one of the most important things for understanding player (in my experience glitch hunting at least) so I think all the state transitions should clear in the code, rather than having hidden "mechanical" action funcs.

Using a function pointer is strange but still I think it's not too different from other actions. As a hypothetical example, suppose we had a function
```
Player_SetupFall(Player* this, PlayState* play, bool shouldRollWhenLanding);
```
which sets up `Player_Action_Fall`, and `Player_Action_Fall` will transition to either `Player_Action_Roll` or `Player_Action_Stand` afterwards depending on what `shouldRollWhenLanding` was. Then in this case the code calling `Player_SetupFall` is mainly controlling the next action after the fall, although it uses a bool instead of a function pointer to do that. But even if `Player_Action_Fall` is very simple I would still consider it to be a normal action rather than a mechanical operation.

Reply:
Yeah sitting on it some more, Im fine with `_SetupWaitForPutAway`, `_Action_WaitForPutAway`, and `afterPutAwayFunc`.
I still think it is more of a "mechanical action" than a normal action like "run" or "fall". But Ill concede that this fact isnt important to factor into the setup action function name

will make the changes momentarily